### PR TITLE
CI/CD for container image

### PR DESCRIPTION
**Pull request by johnandersen777**
- Add Dockerfile
- Add container build to CI
- References
  - https://docs.github.com/en/actions/publishing-packages/publishing-docker-images#publishing-images-to-github-packages
  - https://hub.docker.com/_/python/

```console
$ docker build -t ghcr.io/scitt-community/scitt-api-emulator:main --progress plain .
$ docker run --rm -ti -w /src/src/scitt-api-emulator -v $PWD:/src/src/scitt-api-emulator -p 8000:8000 ghcr.io/scitt-community/scitt-api-emulator:main
```

- Future
  - Enclave (non-virtual CCF) capable containers https://gramine.readthedocs.io/projects/gsc/en/latest/

### CODE REVIEW

**Review by OR13 — DISMISSED** [review]
nice

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -39,3 +39,35 @@ jobs:
         activate-environment: scitt
         environment-file: environment.yml
     - run: python -m pytest
+
+  ci-cd-build-and-push-image-container:
+    name: CI/CD (container)
+    runs-on: ubuntu-latest
+    if: ${{ github.ref == 'refs/heads/main' }}
```
Comment (johnandersen777) [inline .github/workflows/ci.yml]: Force pushed due to originally missing final `}` on this line

**Review by johnandersen777 — COMMENTED** [review]
Realized we should probably build the container on pull requests after seeing and fixing the missing trailing `}` on the `${{ github.ref == 'refs/heads/main' }}` check. This now skips push unless on main branch.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM python:3.8
```
Comment (johnandersen777) [inline Dockerfile:6]: Add usage in comment header

```suggestion
# Usage
#
# Virtual CCF (non-SGX) build and run:
#   $ docker build -t ghcr.io/scitt-community/scitt-api-emulator:main --progress plain .
#   $ docker run --rm -ti -w /src/src/scitt-api-emulator -v $PWD:/src/src/scitt-api-emulator -p 8000:8000 ghcr.io/scitt-community/scitt-api-emulator:main
FROM python:3.8
```

**Review by OR13 — APPROVED** [review]

**Pull request merged**